src/Microsoft.AspNet.OData.Shared/Builder/DerivedTypeConstraintConfiguration.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.OData.Edm.Csdl;
+
+namespace Microsoft.AspNet.OData.Builder
+{
+    /// <summary>
+    /// Configuration for derived type constraints. 
+    /// </summary>
+    public class DerivedTypeConstraintConfiguration
+    {
+        /// <summary>
+        /// Initializes a DerivedTypeConstraintConfiguration instance without setting a base type.
+        /// </summary>
+        internal DerivedTypeConstraintConfiguration()
```

Comment:
>internal [](start = 8, length = 8)

why it's internal?

Reply:
We don't want people to create a DerivedTypeConstraintConfiguration without a base type. 

---
In reply to: [401810329](https://github.com/OData/WebApi/pull/1977#discussion_r401810329) [](ancestors = 401810329)